File: src/runtime/zig/std/time_native.zig
// Zig native implementation for pytra.std.time
// source: src/runtime/zig/std/time_native.zig
const std = @import("std");

/// time.perf_counter() — seconds since arbitrary epoch (nanosecond precision).
pub fn perf_counter() f64 {
    const ns = std.time.nanoTimestamp();
    return @as(f64, @floatFromInt(ns)) / 1_000_000_000.0;
}

File: src/runtime/ruby/std/time_native.rb
# Hand-written native seam for pytra.std.time.
# source: src/pytra/std/time.py
# generated-by: manual (Ruby native runtime)

module PytraNative_Std_Time
  def self.perf_counter
    Process.clock_gettime(Process::CLOCK_MONOTONIC)
  end
end
